DOC: Adds example for using DMSP supporting methods.

rstoneback opened the pull request:
# Description

Addresses #41

Adds examples for multiple DMSP methods. Does not currently have an example for the update ephemeris as we don't currently have a public data source for that data.

# Type of change

Please delete options that are not relevant.

- Documentation Update

# How Has This Been Tested?

No code changes. Visual inspection of output documentation.


## Test Configuration
* Operating system: MacOS
* Version number: Python 3.8

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have linted the files updated in this pull request
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the
release checklist on the pysat wiki:
https://github.com/pysat/pysat/wiki/Checklist-for-Release

rstoneback commented:
I added this example to pysatTutorials as well under 'why not?'. I have an additional hemisphere selection function there that I can bring over here, if desired.

aburrell commented on docs/examples/ex_dmsp_methods.rst:
```diff
@@ -0,0 +1,86 @@
+   ivm.load(date=fdate)
+
+   # Plot multiple orbital passes across polar cap
+   f = plt.figure()
```
```suggestion
   fig = plt.figure()
```

Recommend using more descriptive variable names and saving the single letter variable names for indices.

aburrell commented on docs/examples/ex_dmsp_methods.rst:
```diff
@@ -0,0 +1,86 @@
+   ax.grid(True)
+
+   # Location of spacecraft in relative polar coords for plotting.
+   theta = ivm['mlt'] * (np.pi / 12.) - np.pi / 2.
+   r = 90.- np.abs(ivm['mlat'])
```
```suggestion
   theta = ivm['mlt'] * (np.pi / 12.) - np.pi / 2.0
   r = 90.0- np.abs(ivm['mlat'])
```

Recommend putting the zero in because:
1) It makes it easier to read
2) It makes the number and value of significant figures clear.

aburrell replied:
Also recommend changing `r` to `radius`, see variable name comment above.

aburrell commented on docs/examples/ex_dmsp_methods.rst:
```diff
@@ -0,0 +1,86 @@
+                        clim=[0, 1000.])
+
+   # Plot drifts without both an RPA and DM measurement in red.
+   idx, = np.where(ivm['partial'])
```
Recommend changing `idx` to `ipartial`, which is more descriptive of the data indices being selected.

aburrell commented on docs/examples/ex_dmsp_methods.rst:
```diff
@@ -0,0 +1,86 @@
+
+   # Download data if not already on system
+   if fdate not in ivm.files.files.index:
+       ivm.download(fdate)
```
Need to add user and password kwargs here.

aburrell commented on docs/examples/ex_dmsp_methods.rst:
```diff
@@ -0,0 +1,86 @@
+
+   # Title, colorbar, and save.
+   plt.title('DMSP Ion Drifts')
+   plt.colorbar(vec_plot, label='m/s')
```
Also include variable name in the label, not just units.

aburrell reviewed (changes requested):
I followed the example and found some things that need to be fixed.  More importantly, my figure does not match the example figure.
![pysatMadrigal_dmsp_example](https://user-images.githubusercontent.com/7045886/152207160-a8138bdd-a931-4037-a56b-d417c97c23b1.png)

rstoneback commented:
Thanks @aburrell! I misread one of your comments at first. The upside of that is I added metadata to the DMSP methods. I also added the portion requested once I sorted things out. I'll leave future reviews on this to @JonathonMSmith per previous request.

rstoneback commented:
> I followed the example and found some things that need to be fixed. More importantly, my figure does not match the example figure.
> ![pysatMadrigal_dmsp_example](https://user-images.githubusercontent.com/7045886/152207160-a8138bdd-a931-4037-a56b-d417c97c23b1.png)

I have updated my figure to the correct one. Previous figure had a hemisphere filter on there.

aburrell commented on docs/examples/ex_dmsp_methods.rst:
```diff
@@ -0,0 +1,88 @@
+   # Title, colorbar, and save.
+   plt.title('DMSP Ion Drifts')
+   label = ''.join(['(', ivm.meta['ion_vel_pc_x', ivm.meta.labels.units], ')'])
+   label = ''.join(['Ion Drift ', label])
```
Clarification please! Which ion drift is this?  3D? 2D? cross track?

aburrell replied:
(I think it's the 2D, roughly horizontal velocity, and I think the label or the title should be very clear, since historically labels are sloppy in the literature for DMSP)

aburrell reviewed (commented):
Looks much better!  Since I requested changes on this one, I am willing to see it through.  I'll just need @JonathonMSmith to take over for me on the tutorials. 🐴

rstoneback commented:
> Looks much better! Since I requested changes on this one, I am willing to see it through. I'll just need @JonathonMSmith to take over for me on the tutorials. 🐴

Sounds good. Label updated to read '2D Horizontal Ion Drift'. Could be changed to '2D Roughly Horizontal ...' 🐒

rstoneback commented:
muchas gracias!
